WooCommerce/Classes/ViewModels/Order Details/OrderDetailsDataSource.swift

```diff
@@ -1010,17 +1022,17 @@ extension OrderDetailsDataSource {
                 rows.append(.shippingLabelCreateButton)
             }
 
-            if isProcessingPayment {
+            if isCompletedStatus && shouldShowShippingLabelCreation {
+                rows.append(.shippingLabelCreationInfo(showsSeparator: true))
+            }
```

Comment:
> I'm a little confused by this. Why is the info row only shown when the order has a completed status? Are there other cases where the shipping label button should be shown without that info row?

Yeah, this is mostly a design problem I found while testing different scenarios (ref: p1657690955312339-slack-C0354HSNUJH ), as this label must show below both buttons when both labels are visible. That said, I like your idea of using a switch statement for the different cases, is much more clear how different scenarios play out.

Reply:
Thanks! I see that some of the cases have the same output. One option to simplify that is to declare multiple cases in the same row, for example `case (true, true, _, _), (true, false, false, _):`.

However, since case order matters in `switch` statements, reordering the cases could also simplify the checks you need to make. For example, if you first check the case where an order should show the shipping label and needs processing, then you only need one case definition to cover any other scenarios where an order should show the shipping label.